lib/endpoint-proxy.js

```diff
@@ -521,11 +768,70 @@ class EndpointProxy {
     }
   }
 }
+class ServiceCaller {
+  constructor(client, serviceDeclaration, endpointDeclarations) {
+    if (client === void 0 || client === null) {
+      throw new Error('no database client provided');
+    }
+    if (serviceDeclaration === void 0 || serviceDeclaration === null) {
+      throw new Error('no service declaration provided');
+    }
+    const db = client.getConnectionParams().database;
+    if (db !== void 0 && db !== null) {
+      throw new Error('connection for client cannot specify database');
+    }
+    this._client = client;
+
+  }
+  withFunction(functionDeclaration, moduleExtension) {
+    const overrideExtension = this._moduleExtension;
+    const funcdef = new Functiondef(
+        functionDeclaration,
+        this._endpointDirectory,
+        (overrideExtension !== null) ? overrideExtension : moduleExtension
+    );
+
+    this._functiondefs.set(funcdef.name(), funcdef);
+    return this;
+  }
```

Comment:
The issue was trying to suggest that the constructor should call withFunction() and not that the class needs to provide a withFunction() implementation.